### PR TITLE
Migrated fork from v7.7 to v7.9

### PR DESCRIPTION
**Pull request by christianpaquin**
Merged upstream V_7_9_P1 tag, and updated PQ code as needed.

### CODE REVIEW

**Comment by christianpaquin** [comment]
@dstebila; I didn't test all alg combos; the CircleCI tests should indicate is I missed something. I remember saying something about the latest OpenSSH version using OpenSSL v1.1, but branch 7.9 is still using 1.0.

**Comment by dstebila** [comment]
Triggered integration tests via https://github.com/open-quantum-safe/testing/commits/openssh-79

File: .depend [hunk 1]
```diff
@@ -1,31 +1,31 @@
...
+auth2-pubkey.o: includes.h config.h defines.h platform.h openbsd-compat/openbsd-compat.h openbsd-compat/base64.h openbsd-compat/sigact.h openbsd-compat/readpassphrase.h openbsd-compat/vis.h openbsd-compat/getrrsetbyname.h openbsd-compat/sha1.h openbsd-compat/sha2.h openbsd-compat/rmd160.h openbsd-compat/md5.h openbsd-compat/blf.h openbsd-compat/getopt.h openbsd-compat/bsd-misc.h openbsd-compat/bsd-setres_id.h openbsd-compat/bsd-signal.h openbsd-compat/bsd-statvfs.h openbsd-compat/bsd-waitpid.h openbsd-compat/bsd-poll.h openbsd-compat/fake-rfc2553.h openbsd-compat/bsd-cygwin_util.h openbsd-compat/port-aix.h openbsd-compat/port-irix.h openbsd-compat/port-linux.h openbsd-compat/port-solaris.h openbsd-compat/port-net.h openbsd-compat/port-uw.h openbsd-compat/bsd-nextstep.h entropy.h xmalloc.h ssh.h ssh2.h packet.h openbsd-compat/sys-queue.h dispatch.h opacket.h sshbuf.h log.h misc.h servconf.h compat.h sshkey.h hostfile.h auth.h auth-pam.h audit.h loginrec.h pathnames.h uidswap.h
 auth2-pubkey.o: auth-options.h canohost.h monitor_wrap.h authfile.h match.h ssherr.h channels.h session.h
+<<<<<<< HEAD
```
Comment (dstebila) [inline .depend]: This is a merge artifact that is causing the builds to fail.

Comment (christianpaquin) [reply .depend]: Ah, my `grep *` skipped the dot file. Ok, I'll fix it.

Comment (christianpaquin) [reply .depend]: Just pushed a fix.

**Review by dstebila — APPROVED** [review]

**Pull request merged**